### ISSUE TITLE
Incorrect parsing of float numbers

### ISSUE THREAD

**Issue by loburm** [fluent/fluent-bit#2746]
## Bug Report

**Describe the bug**
Fluent Bit incorrectly parses some float numbers in json logs and stores them as integers. Problem seems to occur when 

**To Reproduce**
```
{"log": 1e-5}
{"log": 1.1e-5}
```
- Steps to reproduce the problem:
Create config in ~/fluentbit-test/conf directory:
```
[SERVICE]
    Log_Level     debug
    Parsers_File  parsers.conf
    HTTP_Server   On
    HTTP_Listen   0.0.0.0
    HTTP_PORT     2020

[INPUT]
    Name   tail
    Tag    test
    Path   /logs/test.log
    Parser json

[OUTPUT]
    Name   stdout
    Match  *
    Format json
```

Create simple parser in the same directory:
```
[PARSER]
    Name        json
    Format      json
```

Start fluent-bit using docker:
```docker run -v ~/fluentbit-test/logs/:/logs -v ~/fluentbit-test/conf:/fluent-bit/etc/ -ti fluent/fluent-bit:1.6 /fluent-bit/bin/fluent-bit -c /fluent-bit/etc/fluent-bit.conf```

Modify file ~/fluentbit-test/logs/test.log and add two lines mentioned above.

**Expected behavior**
fluent bit is suppose to correctly parse both lines and in both cases it should print floats. Instead I'm getting next:

[{"date":1604437986.020481,"val":1},{"date":1604437986.020499,"val":1.1e-05}]
In first log entry val is integer, in second it's float.

**Comment by qingling128** [comment]
@edsiper @fujimotos - Any idea for the expected behavior here? Is this considered a bug with [flb_pack_json](https://github.com/fluent/fluent-bit/blob/d87ee18a22e00ce8d1194707e8285f30deb15ec0/src/flb_pack.c#L219)?

**Comment by edsiper** [comment]
thanks for reporting this; 

yeah, the problem was in the packer, just fixed in 5888f78675a503c4c08f0d87a1f0d8826c62dc98

**Comment by qingling128** [comment]
Thanks a lot @edsiper !
